src/views/studio/modals/transfer-host-confirmation.jsx

```diff
@@ -24,12 +31,45 @@ const TransferHostConfirmation = ({
     const newHostUsername = items.find(item => item.id === selectedId).username;
     const newHostImage = items.find(item => item.id === selectedId).profile.images['90x90'];
     const [passwordInputValue, setPasswordInputValue] = useState('');
+    const [validationError, setValidationError] = useState(null);
+    const {errorAlert, successAlert} = useAlertContext();
+
+    const errorToMessageId = error => {
+        switch (error) {
+        case Errors.RATE_LIMIT: return 'studio.alertTransferRateLimit';
+        case Errors.CONFLICT: return 'studio.transfer.alert.thisUserCannotBecomeHost';
+        default: return 'studio.transfer.alert.somethingWentWrong';
+        }
+    };
+
     const handleSubmit = () => {
-        handleTransfer(passwordInputValue, newHostUsername, selectedId);
+        handleTransferHost(passwordInputValue, newHostUsername, selectedId)
+            .then(() => {
+                handleClose();
+                successAlert({
+                    id: 'studio.alertTransfer',
+                    values: {name: newHostUsername}
+                });
+            })
+            .catch(e => {
+                // For password errors, show validation alert without closing the modal
+                if (e === Errors.PERMISSION) {
```

Comment:
If we add an Errors.PASSWORD, this should check for that

Reply:
to be updated in followup